File: docs/components_page/components/modal/scrollable.R
library(dashBootstrapComponents)
library(dashHtmlComponents)

LOREM <- toString(read.delim(
 "https://raw.githubusercontent.com/facultyai/dash-bootstrap-components/main/docs/components_page/components/modal/lorem.txt",
))

modal_scrollable <- htmlDiv(
  list(
    dbcButton(
      "Scrolling modal", id = "open-scroll", n_clicks = 0, className = "mr-1"
    ),
    dbcButton("Modal with scrollable body", id = "open-body-scroll",
              n_clicks = 0),
    dbcModal(
      list(
        dbcModalHeader("Header"),
        dbcModalBody(LOREM),
        dbcModalFooter(
          dbcButton("Close", id = "close-scroll", n_clicks = 0,
                    className = "ml-auto")
        )
      ),
      id = "modal-scroll",
      is_open = FALSE,
    ),
    dbcModal(
      list(
        dbcModalHeader("Header"),
        dbcModalBody(LOREM),
        dbcModalFooter(
          dbcButton(
            "Close", id = "close-body-scroll", n_clicks = 0,
            className = "ml-auto"
          )
        )
      ),
      id = "modal-body-scroll",
      is_open = FALSE,
      scrollable = TRUE
    )
  )
)


toggle_modal <- function(n1, n2, is_open) {
    if (n1 | n2) {
      return(!is_open)
    }
    return(is_open)
}


app$callback(
  output("modal-scroll", "is_open"),
  list(
    input("open-scroll", "n_clicks"),
    input("close-scroll", "n_clicks"),
    state("modal-scroll", "is_open")
  ), toggle_modal
)

app$callback(
  output("modal-body-scroll", "is_open"),
  list(
    input("open-body-scroll", "n_clicks"),
    input("close-body-scroll", "n_clicks"),
    state("modal-body-scroll", "is_open")
  ), toggle_modal
)

File: docs/components_page/components/modal/scrollable.jl
using DashBootstrapComponents, DashHtmlComponents
using HTTP

LOREM = rstrip(
    String(
        HTTP.get(
            "https://raw.githubusercontent.com/facultyai/dash-bootstrap-components/main/docs/components_page/components/modal/lorem.txt",
        ).body,
    ),
    ['\n'],
)

modal = html_div([
    dbc_button("Scrolling modal", id = "open-scroll", className = "me-1", n_clicks = 0),
    dbc_button("Modal with scrollable body", id = "open-body-scroll", n_clicks = 0),
    dbc_modal(
        [
            dbc_modalheader(dbc_modaltitle("Scrolling modal")),
            dbc_modalbody(LOREM),
            dbc_modalfooter(
                dbc_button(
                    "Close",
                    id = "close-scroll",
                    className = "ms-auto",
                    n_clicks = 0,
                ),
            ),
        ],
        id = "modal-scroll",
        is_open = false,
    ),
    dbc_modal(
        [
            dbc_modalheader(dbc_modaltitle("Modal with scrollable body")),
            dbc_modalbody(LOREM),
            dbc_modalfooter(
                dbc_button(
                    "Close",
                    id = "close-body-scroll",
                    className = "ms-auto",
                    n_clicks = 0,
                ),
            ),
        ],
        id = "modal-body-scroll",
        scrollable = true,
        is_open = false,
    ),
]);

function toggle_modal(n1, n2, is_open)
    return n1 > 0 || n2 > 0 ? is_open == 0 : is_open
end;

callback!(
    app,
    Output("modal-scroll", "is_open"),
    Input("open-scroll", "n_clicks"),
    Input("close-scroll", "n_clicks"),
    State("modal-scroll", "is_open"),
) do n1, n2, is_open
    return toggle_modal(n1, n2, is_open)
end;

callback!(
    app,
    Output("modal-body-scroll", "is_open"),
    Input("open-body-scroll", "n_clicks"),
    Input("close-body-scroll", "n_clicks"),
    State("modal-body-scroll", "is_open"),
) do n1, n2, is_open
    return toggle_modal(n1, n2, is_open)
end;
